external-applications/quarkus-workshop-super-heroes/src/test/java/io/quarkus/ts/openshift/heroes/workshop/VillainsOpenShiftIT.java

```diff
@@ -0,0 +1,177 @@
+package io.quarkus.ts.openshift.heroes.workshop;
+
+import io.quarkus.ts.openshift.common.AdditionalResources;
+import io.quarkus.ts.openshift.common.ManualDeployment;
+import io.quarkus.ts.openshift.common.OpenShiftTest;
+import io.quarkus.ts.openshift.common.injection.TestResource;
+import org.hamcrest.core.Is;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import javax.ws.rs.core.HttpHeaders;
+
+import java.net.URL;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.Response.Status.*;
+import static org.apache.http.HttpHeaders.ACCEPT;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@OpenShiftTest
+@ManualDeployment(appName = "quarkus-workshop-villain", httpRoot = "/", knownEndpoint = "/")
+@AdditionalResources("classpath:openjdk-11-rhel7.yaml")
+@AdditionalResources("classpath:villains-database.yaml")
+@AdditionalResources("classpath:villain.yaml")
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class VillainsOpenShiftIT {
+
+	@TestResource
+	private URL url;
+
+	private static String villainId;
+
+	private static final String DEFAULT_NAME = "Test Villain";
+	private static final String UPDATED_NAME = "Updated Test Villain";
+	private static final String DEFAULT_OTHER_NAME = "Other Test Villain Name";
+	private static final String UPDATED_OTHER_NAME = "Updated Other Test Villain Name";
+	private static final String DEFAULT_PICTURE = "harold.png";
+	private static final String UPDATED_PICTURE = "hackerman.png";
+	private static final String DEFAULT_POWERS = "Partakes in this test";
+	private static final String UPDATED_POWERS = "Partakes in update test";
+	private static final int DEFAULT_LEVEL = 42;
+	private static final int UPDATED_LEVEL = 43;
+
+	@Test
+	public void testHello() {
+		when()
+				.get(url + "/api/villains/hello")
```

Comment:
the offset looks too much, one or additional \tab comparing to `HeroesOpenShiftIT`

Reply:
Yea, tab indentation is evil and should be banned for eternity. Spaces only!

Reply:
This is hopefully fixed too now :)